Add reference page for required permissions to install KIC (#6524)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/reference/required-permissions.md

---

> title: Required Permissions to Use {{ site.kic_product_name }}

Reviewer: ```suggestion
title: Required permissions to use Kong Ingress Controller
```
variables don't work right in front matter, unfortunately:
<img width="719" alt="Screenshot 2023-11-30 at 11 49 04 AM" src="https://github.com/Kong/docs.konghq.com/assets/54370747/66272246-489c-44ee-a1d4-ba1d3f1ea29d">

type: reference
purpose: |

>   What permissions are required to install {{ site.kic_product_name }} if I am not a super admin of the cluster? What permissions are required to run {{ site.kic_product_name }}?

Reviewer: ```suggestion
  What permissions are required to install Kong Ingress Controller if I am not a super admin of the cluster? What permissions are required to run Kong Ingress Controller?
```

---

> To install {{ site.kic_product_name }}, you need to have the permissions to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:

Reviewer 2: ```suggestion
To install {{ site.kic_product_name }}, you need to have the permissions to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) these resources.
```

For creating RBAC rules to enable {{ site.kic_product_name }} to access requried resources, you need to have the permissions to create or update `ClusterRole`, `ClusterRoleBinding` in the cluster scope, and create or update `Role`,`RoleBinding` in the namespace in which you want to install {{ site.kic_product_name }}.

To install CRDs required to configure Kong specific entities, like `KongPlugin` to configure plugins, you need to have the permissions to create and update `CustomResourceDefinition`.

> For creating resources required for {{ site.kic_product_name }}, you need to have the permissions to create or update `Deployment`, `Service`, `ServiceAccount`, `Secret` or `ConfigMap` to create deployments, services and specify their configurations. If you run {{ site.kic_product_name }} with database backed {{site.base_gateway}}, you also need to have the permissions to create or update `PersistentVolumeClaim` to set volumes used for database, and `Job` to run migration jobs. If you want to enable auto scaling, permissions to create or update `HorizontalPodAutoscaler`.

Reviewer: ```suggestion
* For creating resources required for {{ site.kic_product_name }}, you need to have the permissions to create or update `Deployment`, `Service`, `ServiceAccount`, `Secret`,  or `ConfigMap` to create deployments, services, and to specify their configurations. 

* If you run {{ site.kic_product_name }} with database-backed {{site.base_gateway}}, you also need to have the permissions to create or update `PersistentVolumeClaim` to set volumes used for the database, and `Job` to run migration jobs. 

* If you want to enable auto scaling, you also need permissions to create or update `HorizontalPodAutoscaler`.
```

---
title: Required Permissions to Use {{ site.kic_product_name }}
type: reference
purpose: |
  What permissions are required to install {{ site.kic_product_name }} if I am not a super admin of the cluster? What perssions are required to run {{ site.kic_product_name }}?
---

> To install {{ site.kic_product_name }}, you need to have the perssion to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:

Reviewer 3: ```suggestion
To install {{ site.kic_product_name }}, you need to have the permission to WRITE (create, update, get, list, watch in Kubernetes's RBAC model) the following resources:
```

Author: done.

> For creating RBAC rules to enable {{ site.kic_product_name }} to access requried resources, you need to have the permissions to create or update `ClusterRole`, `ClusterRoleBinding` in the cluster scope, and create or update `Role`,`RoleBinding` in the  namespace in which you want to install {{ site.kic_product_name }}.

Author: done.
